src/DTMFSender.ts

```diff
@@ -0,0 +1,124 @@
+class DTMFSender {
+  private _outputNode: MediaStreamAudioDestinationNode;
+  private _outputStream: MediaStream;
+  private _gain: GainNode;
+  private _filter: BiquadFilterNode;
+  private _sourceNode: MediaStreamAudioSourceNode;
+  private _osc1: OscillatorNode;
+  private _osc2: OscillatorNode;
+  private _toneDuration: number = 1500;
+  private _toneGap: number = 70;
+  private _tone: string = "";
+  private _playing: boolean = false;
+
+  private _dtmfFreq: Map<string, Array<number>> = new Map([
+    ["1", [1209, 697]],
+    ["2", [1336, 697]],
+    ["3", [1477, 697]],
+    ["a", [1633, 697]],
+    ["4", [1209, 770]],
+    ["5", [1336, 770]],
+    ["6", [1477, 770]],
+    ["b", [1633, 770]],
+    ["7", [1209, 852]],
+    ["8", [1336, 852]],
+    ["9", [1477, 852]],
+    ["c", [1633, 852]],
+    ["*", [1209, 941]],
+    ["0", [1336, 941]],
+    ["#", [1477, 941]],
+    ["d", [1633, 941]],
```

Comment:
how does someone press 'a', 'b', 'c' or 'd'?

Reply:
Hah, not sure but I got this list from WikiPedia and assumed that there might be some use for the characters as well? I don't have a strong preference though, and can remove them.

https://en.wikipedia.org/wiki/Dual-tone_multi-frequency_signaling

Reply:
I removed them. Will add them back if a requirement comes up later.